### ISSUE TITLE
Lambda class definition

### ISSUE THREAD

**Issue by DaMaoShan** [msmbuilder/vde#13]
self.scale = scale
        self.z_mean = nn.Linear(i, o)
        self.z_log_var = nn.Linear(i, o)

    def forward(self, x):
        self.mu = self.z_mean(x)
        self.log_v = self.z_log_var(x)
        eps = self.scale * Variable(torch.randn(*self.log_v.size())
                                    ).type_as(self.log_v)
        return self.mu + torch.exp(self.log_v / 2.) * eps

In vde class Lambda definition(above):
z_mean = nn.Linear(i, o)
z_log_var = nn.Linear(i, o)

from the variable name:
z_mean is the mean of the input,
z_log_var is the log variation of the input.

**I am trying to understand why they are defined in the same nn function？**
Thank you very much!

**Comment by cxhernandez** [comment]
`nn.Linear` creates an affine transformation layer. The parameters of this affine transformation are updated via optimization of the loss function. So although they're defined similarly in the code, the mean and log-variance functions are in fact very different functions after training.

Let me know if this clarifies anything or if I'm misinterpreting your question.

**Comment by DaMaoShan** [comment]
Thanks for your reply.
 And which is the learnable parameters in nn.Linear?
OR, the learning mechanism is constructed in torch.nn?

**Comment by cxhernandez** [comment]
For that I'd refer you to the [PyTorch documentation](https://pytorch.org/tutorials/beginner/pytorch_with_examples.html#).

**Comment by DaMaoShan** [comment]
Thank you
